### PR TITLE
feat(core): add logic to compose and send email

### PR DESCRIPTION
**Pull request by wrmk**
Part of #262.

all logic was copied from the internal project and adapted to freeats

we may discuss this pr via call to make your review easier

- [x] I have reviewed my code in "Files changed" tab.
- [x] I have re-read the issue and this PR fully resolves it.

### CODE REVIEW

File: app/assets/javascript/controllers/compose_controller.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  static outlets = [
+    "button-utils",
+  ];
+
+  newThreadFormTargetConnected() {
+    this.#setupComposeInterruptHandler();
+    this.initialSubjectContent = this.formSubjectTarget.value;
+    this.buttonUtilsOutlets.forEach((btn) =>
+      // Use locale!!!!!!
```
Comment (wrmk) [inline app/assets/javascript/controllers/compose_controller.js:21]: !!!!use locale!!!!
and below too

**Comment by wrmk** [comment]
!!need to create issue about signing out even alert about unsaved changes is displayed!!

File: app/views/ats/email_messages/_email_message.html.slim
Comment (wrmk) [inline app/views/ats/email_messages/_email_message.html.slim:1]: unused

File: config/environments/development.rb [hunk 2]
```diff
@@ -91,7 +91,8 @@
     port: ENV.fetch("MAILER_SMTP_PORT", "").to_i,
     domain: ENV.fetch("MAILER_SMTP_DOMAIN", ""),
     authentication: ENV.fetch("MAILER_SMTP_AUTHENTICATION", ""),
-    enable_starttls_auto: ENV.fetch("MAILER_SMTP_ENABLE_STARTTLS_AUTO", "true") == "true"
+    enable_starttls_auto: ENV.fetch("MAILER_SMTP_ENABLE_STARTTLS_AUTO", "true") == "true",
+    return_response: true
```
Comment (wrmk) [inline config/environments/development.rb:95]: need to get response from smtp mailer

File: app/controllers/ats/compose_controller.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  def create
+    email_message_params = compose_email_message_params
+    validation = EmailMessageSchema.new.call(email_message_params.compact)
+    if validation.errors.present?
+      render_error schema_errors_to_string(validation.errors), status: :unprocessable_entity
+      return
+    end
+
+    email_addresses = email_message_params[:to].join(", ")
+    result = EmailMessageMailer.with(email_message_params).send_email.deliver_now!
```
Comment (wrmk) [inline app/controllers/ats/compose_controller.rb:32]: without '!" we did not get response

File: app/controllers/ats/compose_controller.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    if (result.is_a?(Net::SMTP::Response) && result.status == "250") ||
```
Comment (wrmk) [inline app/controllers/ats/compose_controller.rb:34]: this line works on prod

File: app/controllers/ats/compose_controller.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+       (result.is_a?(Mail::Message) && !Rails.env.production?)
```
Comment (wrmk) [inline app/controllers/ats/compose_controller.rb:35]: this line works on development and test

File: test/components/previews/button_link_component_preview.rb [hunk 6]
```diff
@@ -96,7 +96,7 @@ def with_tooltip(
   def disabled_with_tooltip(
     icon_name: :arrow_right,
     size: :medium,
-    disabled: false,
+    disabled: true,
```
Comment (wrmk) [inline test/components/previews/button_link_component_preview.rb:99]: fixed typo, button should be disabled in preview

**Comment by wrmk** [comment]
> !!need to create issue about signing out even alert about unsaved changes is displayed!!

https://github.com/freeats/freeats/issues/274

File: app/controllers/ats/compose_controller.rb [hunk 7]
```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class ATS::ComposeController < AuthorizedController
+  include SchemaHelper
+  before_action { authorize! :compose }
+
+  FROM_ADDRESS = "notifications@freeats.com"
```
Comment (wrmk) [inline app/controllers/ats/compose_controller.rb:7]: use env fetch

**Comment by Lyams** [comment]
Screenshot from 2024-11-29 11-42-12

I suggest to remove emails counter in this PR too because we do not use related code.

**Comment by Lyams** [comment]
When selecting email addresses in the bсс and сс fields, we can select emails of deactivated members. How correct is this?
Screenshot from 2024-11-29 11-45-14

File: app/controllers/ats/compose_controller.rb [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+class ATS::ComposeController < AuthorizedController
+  include SchemaHelper
+  before_action { authorize! :compose }
+
+  FROM_ADDRESS = ENV.fetch("NOTIFICATION_ADDRESS", nil)
+
+  def new
+    candidate = Candidate.find(params[:candidate_id])
+    candidate_email_addresses = candidate.all_emails
+    members_email_addresses = Member.email_addresses(except: current_member)
```
Comment (Lyams) [inline app/controllers/ats/compose_controller.rb:12]: I think that deactivated members should also be excluded.

**Review by Lyams — APPROVED** [review]
Good work :mortar_board:

**Pull request merged**